lib/recipe/widgets/recipe_ingredients_screen.dart

```diff
@@ -76,19 +76,21 @@ class RecipeNameInput extends StatelessWidget {
     return BlocBuilder<RecipeBloc, RecipeState>(
       builder: (context, state) {
         return TextFormField(
-          initialValue: state.recipes[recipeIndex].recipeName.value,
+          initialValue: state.recipes[recipeIndex].recipeName,
           decoration: InputDecoration(
             icon: const Icon(Icons.assignment_rounded),
             labelText: 'Recipe Name',
             helperText: 'A valid recipe name, e.g. Aloo bandhgobhi',
-            errorText: state.recipes[recipeIndex].recipeName.invalid
+            errorText: (state.recipes[recipeIndex].recipeName != null &&
+                    state.recipes[recipeIndex].recipeName == '')
```

Comment:
Again I think it would be a little nicer to have it named expressively and refactored into a method.